californium-core/src/test/java/org/eclipse/californium/core/test/LinkFormatTest.java

```diff
@@ -0,0 +1,334 @@
+package org.eclipse.californium.core.test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import org.eclipse.californium.category.Small;
+import org.eclipse.californium.core.CoapResource;
+import org.eclipse.californium.core.coap.CoAP.Code;
+import org.eclipse.californium.core.coap.LinkFormat;
+import org.eclipse.californium.core.coap.Request;
+import org.eclipse.californium.core.server.resources.Resource;
+import org.eclipse.californium.core.server.resources.ResourceAttributes;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.Assert;
+import org.junit.experimental.categories.Category;
+
+@Category(Small.class)
+public class LinkFormatTest {
+	private Resource root;
+	private Request request1, request2, request3;
+
+	@Before
+	public void setup() {
+		root = new CoapResource("");
+		Resource sensors = new CoapResource("sensors");
+		Resource actors = new CoapResource("actors");
+		Resource temp = new CoapResource("temp");
+		Resource light = new CoapResource("light");
+		Resource blinds = new CoapResource("blinds");
+
+		root.add(sensors);
+		root.add(actors);
+
+		sensors.add(temp);
+		sensors.add(light);
+		sensors.getAttributes().setTitle("Sensor Index");
+
+		actors.add(blinds);
+		actors.getAttributes().setTitle("Actor Index");
+
+		temp.getAttributes().addResourceType("temperature-c");
+		temp.getAttributes().addInterfaceDescription("sensor");
+		temp.getAttributes().addAttribute("foo");
+		temp.getAttributes().addAttribute("bar", "one");
+		temp.getAttributes().addAttribute("bar", "two");
+
+		light.getAttributes().addResourceType("light-lux");
+		light.getAttributes().addInterfaceDescription("sensor");
+
+		blinds.getAttributes().addResourceType("angle");
+		blinds.getAttributes().addInterfaceDescription("actor");
+
+		request1 = new Request(Code.GET);
+		request1.setURI("/.well-known/core?if=sensor");
+		request2 = new Request(Code.GET);
+		request2.setURI("/.well-known/core?if=sensor&rt=light-lux");
+		request3 = new Request(Code.GET);
+		request3.setURI("/.well-known/core?if=sensor&rt=angle");
+
+	}
+
+	@Test
+	public void testMatchesFixed() {
```

Comment:
Can you find a more descriptive name indicating what is to be tested/asserted, e.g. something like `testMatchesCorrectlyRecognizesBlaBlaBla...`? Alternatively, please add a JavaDoc shortly describing what the test validates/asserts. This makes it much easier for others to a) understand what the purpose of the test is and b) refactor the code if necessary when introducing changes in other places.